diag_manager/fms_diag_object.F90

```diff
@@ -540,6 +540,10 @@ logical function fms_diag_accept_data (this, diag_field_id, field_data, mask, rm
   !< Set the field_weight. If "weight" is not present it will be set to 1.0_r8_kind
   field_weight = set_weight(weight)
 
+  !< Set the variable type based off passed in field data
+  if(.not. this%FMS_diag_fields(diag_field_id)%has_vartype()) &
+    call this%FMS_diag_fields(diag_field_id)%set_type(field_data(1,1,1,1))
+
```

Comment:
This is already done inside the OMP critical section:
https://github.com/NOAA-GFDL/FMS/blob/0c2f6802c531ae3c4c48c249c96d4b803a61fbdd/diag_manager/fms_diag_object.F90#L604-L656

Reply:
removed with da79b88